src/shared/json.h

```diff
@@ -290,6 +291,8 @@ enum {
         _JSON_BUILD_MAX,
 };
 
+typedef int (*JsonBuildCallback)(const char *name, void *userdata, JsonVariant **ret);
```

Comment:
we almost always put the userdata pointer last, i think we should do so here too